biggraphite/cli/command_stats.py

```diff
@@ -117,13 +120,21 @@ def run(self, accessor, opts):
         """
         self.ns = Namespaces(opts.conf)
         accessor.connect()
-        accessor.map(
-            self.stats,
-            start_key=opts.start_key,
-            end_key=opts.end_key,
-            shard=opts.shard,
-            nshards=opts.nshards,
-        )
+
+        if accessor.TYPE.startswith("elasticsearch+"):
+            accessor = accessor._metadata_accessor
+
+        if accessor.TYPE == "elasticsearch":
+            # Elasticsearch has a better implementation.
```

Comment:
Can't we call a generic stats function, that in cassandra implementation would call map() as needed ?

Reply:
No, because in most implementations we iterate on metrics. For ES we iterate on patterns.

Reply:
ok